### DISCUSSION TITLE
Inclusion of webr extension in "base" Quarto?

### DISCUSSION THREAD

**Discussion by melissavanbussel** [quarto-dev/quarto-cli#7160] [Feature Requests]
I'm not really sure if this is a question or if it should be a feature request, but I'm wondering if there are plans to eventually include the [webr Quarto extension](https://github.com/coatless/quarto-webr) as part of the "base" Quarto so that the extension wouldn't have to be added separately to each project by the user? 

I ask because I would love to use this extension at work, but am currently unable to (as we aren't allowed to install anything directly from GitHub). We **are** able, however, to use any version of Quarto (including pre-release). I suspect there may be other workplaces that would face this same challenge (i.e., *Quarto* is on the list of approved software, but extensions from GitHub are not). 

If this isn't possible, are there any alternative approaches that might work?

Thank you so much for your help!

**Comment by mcanouil** [comment]
Some features started as extensions and were embedded in Quarto CLI.
For instance, lightbox extension is currently being embedded inside Quarto CLI.

This being said, it's really on a case by case basis especially when the extensions are community based like the `quarto-webr` extension.

It is possible that in the future such extensions will be embedded but there are no plans defining which will be or not, or even when.

Note that depending on the reason for which extensions/GitHub repositories are not allowed in your workplace you could argue to have some exceptions for some extensions.
From a security standpoint, since Quarto allows to have any code running in it, the extension is not less "secure" as the "security breach" would actually be the users.

**Reply by melissavanbussel** [reply]
Thanks @mcanouil for explaining, that makes sense. Is this something that would be well suited to a feature request then?

**Reply by mcanouil** [reply]
I changed the category but don’t expect  anything here.

**Comment by cderv** [comment]
> From a security standpoint, since Quarto allows to have any code running in it, the extension is not less "secure" as the "security breach" would actually be the users.

I would add to this that an extension for Quarto is only a set of files that needs to be organized a certain way for the project. Github is only a way to distribute the extension, and Quarto handles installation easily. However, an extension can be installed from a local bundle for example (https://quarto.org/docs/extensions/distributing.html#archive-files). 

Maybe this information can help you see with your IT to have available a version of this bundle in your network to install it easily in your projects. Unless you don't need any exception for downloading and unzipping file from the web. 

Just wanted to clarify this in case it was not. Thanks
